### PR TITLE
Add adapter for YouTrack

### PR DESCRIPTION
**Pull request by pmeinhardt**
Heya! 👋 I hope you're all doing great.

Here's a new adapter for [YouTrack](https://www.jetbrains.com/youtrack/).

_Disclaimer: I haven't tested this on a public cloud instance, configuration and markup might look slightly different there. But I guess we can improve support in case issues about this adapter do come in._

### CODE REVIEW

File: src/core/adapters/youtrack.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import { $has, $text } from "./dom-helpers";
+import { hasRequiredDetails } from "./utils";
+
+async function scan(url: URL, document: Document): Promise<TicketData[]> {
+  if (!$has(".yt-issue-view", document)) return [];
+
+  const id = $text(".js-issue-id", document)?.trim();
+  const title = $text("[data-test='issueSummary']", document);
+  const type =
+    $text("[data-test='Type']", document)?.toLowerCase().trim() || "feature";
```
Comment (klappradla) [inline src/core/adapters/youtrack.ts:17]: Weird that the word "type" is capitalized here. But you'll probably notice quite fast when then changes...

File: src/core/adapters/youtrack.test.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
+/**
+ * @jest-environment node
+ */
+
+import { JSDOM } from "jsdom";
+
+import scan from "./youtrack";
+
+const html = {
+  issuepage: (type?: string) => `
```
Comment (klappradla) [inline src/core/adapters/youtrack.test.ts:10]: Is this structure supposed to leave room for quickly adding support for other pages in the future? Sweet 🍬 

From the product page it looks like there's a Jira-like backlog view where one quick-views issues. That could could probably be a candidate 👀

**Review by klappradla — APPROVED** [review]
Nice one 🍭 

Always in for a good old DOM-adapter 🙌. Didn't know this YT-tool existed 💡

Solid implementation as always 🪨. I briefly wondered about the early return in `scan`, because the CSS selector looks generic enough to be present on other tracker pages as well. But with checking the URL not being an option and scanning with this adapter being pretty fast, I'd just roll with it like it is.

File: src/core/adapters/youtrack.test.ts (hunk 2 above)
Comment (pmeinhardt) [reply src/core/adapters/youtrack.test.ts:10]: Yep, exactly. I took inspiration from the existing Jira adapter when I whipped up this PR. It seemed like there might be other pages to extract tickets from and/or subtle differences between the internal JetBrains YouTrack instance and cloud-hosted instances. I can imagine that there's on-premise instances out there too… To get started I decided to just go for this one case.

**Pull request merged**